### ISSUE TITLE
Build Failure on Android

### ISSUE THREAD

**Issue by xeinebiu** [authpass/biometric_storage#58]
````

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':biometric_storage:kaptGenerateStubsDebugKotlin'.
> Could not resolve all files for configuration ':biometric_storage:debugCompileClasspath'.
   > Failed to transform moshi-1.13.0.jar (com.squareup.moshi:moshi:1.13.0) to match attributes {artifactType=android-classes-jar, org.gradle.category=library, org.gradle.dependency.bundling=external, org.gradle.jvm.environment=standard-jvm, org.gradle.jvm.version=8, org.gradle.libraryelements=jar, org.gradle.status=release, org.gradle.usage=java-api, org.jetbrains.kotlin.platform.type=jvm}.
      > Execution failed for JetifyTransform: C:\Users\xeine\.gradle\caches\modules-2\files-2.1\com.squareup.moshi\moshi\1.13.0\da685586facab9eb5c4fb630ce248be14e7da21b\moshi-1.13.0.jar.
         > Failed to transform 'C:\Users\xeine\.gradle\caches\modules-2\files-2.1\com.squareup.moshi\moshi\1.13.0\da685586facab9eb5c4fb630ce248be14e7da21b\moshi-1.13.0.jar' using Jetifier. Reason: IllegalArgumentException, message: Unsupported class file major version 60. (Run with --stacktrace for more details.)
           Suggestions:
            - Check out existing issues at https://issuetracker.google.com/issues?q=componentid:460323&s=modified_time:desc, it's possible that this issue has already been filed there.
            - If this issue has not been filed, please report it at https://issuetracker.google.com/issues/new?component=460323 (run with --stacktrace and provide a stack trace if possible).

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 23s
Exception: Gradle task assembleDebug failed with exit code 1


````

### Flutter Doctor
````

[√] Flutter (Channel stable, 2.10.3, on Microsoft Windows [Version 10.0.22000.588], locale en-DE)
    • Flutter version 2.10.3 at C:\tools\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (3 weeks ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\xeine\AppData\Local\Android\Sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_SDK_ROOT = C:\Users\xeine\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop for Windows
    X Visual Studio not installed; this is necessary for Windows development.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[√] Android Studio (version 2021.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22000.588]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 99.0.4844.74
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 97.0.1072.69

[√] HTTP Host Availability
    • All required HTTP hosts are available

````

**Comment by MyBuzzTechnologies** [comment]
I'm also having this problem building on Android with the same flutter version (2.10.3).
Any suggestions?

**Comment by EhabSalah** [comment]
Use the version before the last one `biometric_storage: 4.0.0-dev.1`

**Comment by tahitiangabriel** [comment]
This is a duplicate of #55

**Comment by hpoul** [comment]
Can you please try out `4.1.0-dev.1` - i've now simply removed the moshi dependency altogether.. there seems to be some weird bugs somewhere... https://issuetracker.google.com/issues/216429430 https://issuetracker.google.com/issues/159151549 .. and Those two classes aren't really hard to deserialize without some fancy json library 😅

**Comment by xeinebiu** [comment]
> Can you please try out `4.1.0-dev.1` - i've now simply removed the moshi dependency altogether.. there seems to be some weird bugs somewhere... https://issuetracker.google.com/issues/216429430 https://issuetracker.google.com/issues/159151549 .. and Those two classes aren't really hard to deserialize without some fancy json library 😅

  biometric_storage: 4.0.0-dev.1

````
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':biometric_storage:kaptReleaseKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.gradle.internal.KaptWithoutKotlincTask$KaptExecutionWorkAction
   > java.lang.reflect.InvocationTargetException (no error message)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

````

**Comment by hpoul** [comment]
> biometric_storage: 4.0.0-dev.1

Is that a typo? 4.1.0-dev.1 is the new version https://pub.dev/packages/biometric_storage/versions/4.1.0-dev.1

**Comment by tahitiangabriel** [comment]
> Can you please try out `4.1.0-dev.1` - i've now simply removed the moshi dependency altogether.. there seems to be some weird bugs somewhere... https://issuetracker.google.com/issues/216429430 https://issuetracker.google.com/issues/159151549 .. and Those two classes aren't really hard to deserialize without some fancy json library 😅

It works with this version. 👍

**Comment by xeinebiu** [comment]
> > biometric_storage: 4.0.0-dev.1
> 
> Is that a typo? 4.1.0-dev.1 is the new version https://pub.dev/packages/biometric_storage/versions/4.1.0-dev.1

Looks like I used the version provided by @EhabSalah

**Comment by hpoul** [comment]
> It works with this version. 👍

@tahitiangabriel thanks for testing.. i've released `4.1.0`, let's hope this is solved now 😂

**Comment by tahitiangabriel** [comment]
@hpoul I think you can also close #55
